### PR TITLE
Adds OSS::CurrencyInput component

### PR DESCRIPTION
**Pull request by Miexil**
### What does this PR do?

Adds OSS::CurrencyInput component

### Good PR checklist
- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/currency-input.hbs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    <div class="currency-selector fx-row" role="button" {{on "click" this.toggleCurrencySelector}}>
+      <span class="margin-right-xxx-sm text-center symbol">{{this.selectedCurrencySymbol}}</span>
+      {{#if this.currencySelectorShown}}
+        <i class="fal fa-chevron-up margin-left-xxx-sm"></i>
+      {{else}}
+        <i class="fal fa-chevron-down margin-left-xxx-sm"></i>
+      {{/if}}
+    </div>
+    <Input class="fx-1" type="tel" @value={{this.value}} {{on "keydown" this.onlyNumeric}}
+           placeholder="81726354" autocomplete="off" />
```
Comment (phndiaye) [inline addon/components/o-s-s/currency-input.hbs:12]: sure you want to keep this placeholder? :p

File: addon/components/o-s-s/currency-input.stories.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        type: {
+          summary: 'number'
+        },
+        defaultValue: { summary: false }
+      },
+      control: { type: 'text' }
+    },
+    onChange: {
+      description: 'A callback that sends the modifications of the value & the currency back to the parent component'
+    }
```
Comment (JulienVannier66) [inline addon/components/o-s-s/currency-input.stories.js:30]:
```suggestion
    type: { required: true },
    onChange: {
      description: 'A callback that sends the modifications of the value & the currency back to the parent component'
    }
```
To set "*" directly in the doc

File: addon/components/o-s-s/currency-input.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  }
+
+  get selectedCurrencySymbol(): string {
+    return this.selectedCurrency.symbol || '—';
+  }
+
+  @action
+  onlyNumeric(event: KeyboardEvent): void {
+    const authorizedInputs = ['Backspace', 'Delete', 'ArrowLeft', 'ArrowRight', 'Tab', 'Shift', 'Control', '.', ','];
+    if (/^[0-9]$/i.test(event.key) || authorizedInputs.find((key: string) => key === event.key)) {
```
Comment (JulienVannier66) [inline addon/components/o-s-s/currency-input.ts:42]: Set regex in a const?

File: addon/components/o-s-s/currency-input.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    this.filteredCurrencies = this._currencies;
+  }
+}
+
+type Currency = {
+  code: string;
+  symbol: string;
+};
+
+const usedCurrencies: Currency[] = [
```
Comment (JulienVannier66) [inline addon/components/o-s-s/currency-input.ts:83]: It is pertinent to set this const in utils or somethink like that?

File: app/styles/currency-input.less [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  .currency-input {
+    .currency-selector {
+      border-right: 1px solid @border-color;
+      height: 100%;
+      display: flex;
+      align-items: center;
+      padding-right: 12px;
+      margin-right: 12px;
```
Comment (JulienVannier66) [inline app/styles/currency-input.less:31]:
```suggestion
      padding-right: var(--spacing-px-12);
      margin-right: var(--spacing-px-12);
```

**Review by phndiaye — APPROVED** [review]

File: app/styles/currency-input.less [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+      background-color: transparent;
+    }
+
+    input:focus {
+      border: none;
+      outline: none;
+    }
+
+    &:focus-within {
+      border: 1px solid @upf-gray;
```
Comment (JulienVannier66) [inline app/styles/currency-input.less:51]:
```suggestion
      border: 1px solid var(--color-border-default);
```
No good with this error

File: app/styles/currency-input.less [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  .symbol {
+    width: 22px;
+    font-weight: bold;
+  }
+
+  .currency-input {
+    .currency-selector {
+      border-right: 1px solid @border-color;
```
[the remark this answers is not in the document]
Comment (Miexil) [reply app/styles/currency-input.less:26]: Done !

File: addon/components/o-s-s/currency-input.ts (hunk 4 above)
Comment (Miexil) [reply addon/components/o-s-s/currency-input.ts:83]: No use for now, we're not using this set anywhere else !

File: app/styles/currency-input.less [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  .currency-input {
+    .currency-selector {
+      border-right: 1px solid var(--color-border-default);
+      height: 100%;
+      display: flex;
+      align-items: center;
+      padding-right: 12px;
+      margin-right: 12px;
```
Comment (JulienVannier66) [inline app/styles/currency-input.less:31]: If css var doesn't work use `@spacing-xx-sm`. Otherwise LGTM

**Review by JulienVannier66 — APPROVED** [review]

File: app/styles/currency-input.less (hunk 8 above)
Comment (Miexil) [reply app/styles/currency-input.less]: Fixed!

**Review by JulienVannier66 — APPROVED** [review]

**Review by phndiaye — APPROVED** [review]

**Pull request merged**